### PR TITLE
Feat/output endpoints generic bpm

### PR DESCRIPTION
**Pull request by rainer-exxcellent**
## Description
Created endpoints for uploading and searching generic business partner output data.

### Solves

#387

### CODE REVIEW

File: bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/GateBusinessPartnerApi.kt [hunk 1]
```diff
@@ -76,4 +78,23 @@ interface GateBusinessPartnerApi {
...
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "The requested page of business partners"),
+            ApiResponse(responseCode = "400", description = "On malformed pagination request", content = [Content()]),
+        ]
+    )
+    @PostMapping("/output/business-partners/search")
+    @PostExchange("/output/business-partners/search")
+    fun getLegalEntitiesOutput(
```
Comment (martinfkaeser) [inline bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/GateBusinessPartnerApi.kt:95]: Consistent naming: Rename to `getBusinessPartnersOutputByExternalIds`?

**Comment by martinfkaeser** [comment]
I would suggest renaming the interface / data classes:
- IBaseBusinessPartnerInputDto -> IBaseBusinessPartnerDto (used as basis for both input and output)
- BusinessPartnerInputDto -> BusinessPartnerInputResponse
- BusinessPartnerOutputDto -> BusinessPartnerOutputResponse

**Comment by rainer-exxcellent** [comment]
> I would suggest renaming the interface / data classes:
> 
>     * IBaseBusinessPartnerInputDto -> IBaseBusinessPartnerDto (used as basis for both input and output)
> 
>     * BusinessPartnerInputDto -> BusinessPartnerInputResponse
> 
>     * BusinessPartnerOutputDto -> BusinessPartnerOutputResponse

Most of the class in the response package end with Dto. So it is named consistently.
We should first discuss about the naming before we change this.

**Comment by martinfkaeser** [comment]
> > I would suggest renaming the interface / data classes:
> > ```
> > * IBaseBusinessPartnerInputDto -> IBaseBusinessPartnerDto (used as basis for both input and output)
> > 
> > * BusinessPartnerInputDto -> BusinessPartnerInputResponse
> > 
> > * BusinessPartnerOutputDto -> BusinessPartnerOutputResponse
> > ```
> 
> Most of the class in the response package end with Dto. So it is named consistently. We should first discuss about the naming before we change this.

I would suggest naming the DTOs that are only used in a request `...Request` (and put them in the request package), the DTOs that are only used in a response `...Response` (and put them in the response package) and the ones which are used in both contexts `...Dto`.
All the DTOs in the gate/request package are consistently named `...Request`.
I don't know why there is a mix of `...Dto` and `...Response` in the gate/response package, but feels inconsistent.

**Review by nicoprow — CHANGES_REQUESTED** [review]
Code looks good to me.

Please squash the commits in the history. They only contain rebase fixes and adjustments from review.

**Review by nicoprow — APPROVED** [review]

**Pull request merged**